src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+                neurodata_type_inc="VectorData",
+                doc="Width of a rectangular or square contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="height_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Height of a rectangular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+        ],
+    )
+
+    probe = NWBGroupSpec(
+        neurodata_type_def="Probe",
+        neurodata_type_inc="Device",
+        doc="Specific instance of a neural probe object.",
+        links=[
+            NWBLinkSpec(
+                name="probe_model",
+                doc="The model of the probe used to record the data.",
+                target_type="ProbeModel",
+            ),
+        ],
+        attributes=[
+            NWBAttributeSpec(
+                name="identifier",
+                doc="Identifier of the probe, usually the serial number.",
+                dtype="text",
+                required=False,
+            ),
+        ],
+    )
+
+    probe_model = NWBGroupSpec(
+        neurodata_type_def="ProbeModel",
+        neurodata_type_inc="Device",
+        doc=(
+            "Neural probe object, compatible with the ProbeInterface specification. The name of the object should "
+            'be the model name of the probe, e.g., "Neuropixels 1.0".'
+        ),
+        groups=[
+            NWBGroupSpec(
+                name="contacts_table",
+                neurodata_type_inc="ContactsTable",
+                doc="Neural probe contacts, compatible with the ProbeInterface specification",
+            ),
+        ],
+        attributes=[
+            # inherits name, description, manufacturer from Device
+            NWBAttributeSpec(name="ndim", doc="dimension of the probe", dtype="int", default_value=2),
+            NWBAttributeSpec(
+                # although the ProbeModel also has a name attribute, the name must be unique across all
+                # devices in the NWB file, and users may decide to use a more descriptive name than just
+                # the model name
+                name="model",
+                doc='Name of the model of the probe, e.g., "Neuropixels 1.0".',
+                dtype="text",
+            ),
+            NWBAttributeSpec(
+                name="planar_contour_in_um",  # TODO should this just be "contour"?
+                doc=(
+                    "The coordinates of the nodes of the polygon that describe the shape (contour) of the probe, "
+                    "in micrometers. The first and last points are connected to close the polygon. "
+                    "e.g., [(-20., -30.), (20., -110.), (60., -30.), (60., 190.), (-20., 190.)]."
+                    "See 'probe_planar_contour' in "
+                    "https://probeinterface.readthedocs.io/en/main/format_spec.html for more details."
```

Comment:
This link is not very clarifying but I think this is on probeinterface. We should probably a better explanation there.

Reply:
I agree. I included the link to indicate that we intend this field to map to that field. I think the description here is pretty good, but please feel free to improve it here (or there).

Reply:
The description is here is good. It is on not good on probeinterface side.